lib/src/test/java/com/knuddels/jtokkit/BaseEncodingRegistryTest.java

```diff
@@ -6,110 +6,110 @@
 
 import java.util.Collections;
 import java.util.List;
-import java.util.Optional;
 import java.util.function.Consumer;
 import java.util.regex.Pattern;
 
 import static org.junit.jupiter.api.Assertions.*;
 
-public abstract class BaseEncodingRegistryTest<T extends AbstractEncodingRegistry> {
+abstract class BaseEncodingRegistryTest<T extends AbstractEncodingRegistry> {
 
     protected final T registry;
     protected final Consumer<T> initializer;
 
-    public BaseEncodingRegistryTest(final T registry) {
-        this(registry, __ -> {});
+    BaseEncodingRegistryTest(T registry) {
+        this(registry, __ -> {
+        });
     }
 
-    public BaseEncodingRegistryTest(final T registry, final Consumer<T> initializer) {
+    BaseEncodingRegistryTest(T registry, Consumer<T> initializer) {
         this.registry = registry;
         this.initializer = initializer;
     }
 
     @BeforeEach
-    public void setup() {
+    void setup() {
         initializer.accept(registry);
     }
 
     @Test
-    public void getEncodingReturnsCorrectEncoding() {
-        for (final EncodingType type : EncodingType.values()) {
-            final Encoding encoding = registry.getEncoding(type);
+    void getEncodingReturnsCorrectEncoding() {
+        for (var type : EncodingType.values()) {
```

Comment:
if we decide to run the tests on JDK 8 as well, we will have to add the types back

Reply:
as mentioned above: let's cross this bridge after the main optimization PRs are done